python/tskit/tables.py

```diff
@@ -582,6 +582,20 @@ def append(self, row):
             }
         )
 
+    def replace_with(self, other):
+        """
+        Entirely replace the contents of this table with another
+        """
+        if type(self) != type(other):
```

Comment:
I think this is overly restrictive, and prevents accepting anything that might have the same columns but isn't the same type, for example a subclass.
I think we should instead try to get each column and give a nice error if `other` doesn't have the attribute.